### PR TITLE
Feature errors cleanup

### PR DESCRIPTION
**Pull request by holashchand**

### CODE REVIEW

**Review by Sreejit-K — CHANGES_REQUESTED** [review]

File: .github/workflows/maven.yml
[the remark this answers is not in the document]
Comment (holashchand) [reply .github/workflows/maven.yml:1]: it is for github actions, I have left commented debug code to debug in future

File: deps/keycloak-mobile-number-login-spi/src/main/java/in/divoc/api/authenticator/NotifyService.java [hunk 1]
```diff
@@ -44,7 +45,7 @@ public void notify(String to, String otp) throws IOException {
 				System.out.println(response.toString());
 			}
 		} catch (IOException e) {
-			logger.error(e);
+			logger.error("Exception occurred while notifying: {}", e.getMessage());
```
Comment (srprasanna) [inline deps/keycloak-mobile-number-login-spi/src/main/java/in/divoc/api/authenticator/NotifyService.java:48]: can you use ExceptionUtils to print stack trace. throw the exception

File: docker-compose.yml [hunk 2]
```diff
@@ -54,15 +55,18 @@ services:
...
       - sunbird_sso_client_id=${KEYCLOAK_CLIENT_ID-registry-frontend}
       - sunbird_sso_admin_client_secret=${KEYCLOAK_SECRET}
+      - claims_enabled=${CLAIMS_ENABLED-false}
       - claims_url=http://claim-ms:8082
+      - signature_enabled=${SIGNATURE_ENABLED-false}
       - sign_url=http://certificate-signer:8079/sign
       - verify_url=http://certificate-signer:8079/verify
       - sign_health_check_url=http://certificate-signer:8079/health
-      - signature_enabled=true
+      - certificate_enabled=${CERTIFICATE_ENABLED-false}
```
Comment (srprasanna) [inline docker-compose.yml:64]: can you modify the microsite with this as well?

File: docker-compose.yml [hunk 3]
```diff
@@ -213,7 +217,7 @@ services:
       zookeeper:
         condition: service_started
     ports:
-      - "9092:9092"
+      - "9094:9094"
```
Comment (srprasanna) [inline docker-compose.yml:220]: Keep it as 9092 only, that is the default.

File: java/claim/src/main/java/dev/sunbirdrc/claim/service/ClaimsAuthorizer.java [hunk 4]
```diff
@@ -35,7 +35,7 @@ public boolean isAuthorizedAttestor(Claim claim, JsonNode attestorNode) {
             );
             return conditionResolverService.evaluate(resolvedCondition);
         } catch (Exception e) {
-            logger.error(e.getMessage());
+            logger.error("Exception occurred while resolving condition {}", e.getMessage());
```
Comment (srprasanna) [inline java/claim/src/main/java/dev/sunbirdrc/claim/service/ClaimsAuthorizer.java:38]: Log the exception using ExceptionUtils.

File: java/elastic-search/src/main/java/dev/sunbirdrc/elastic/ElasticServiceImpl.java [hunk 5]
```diff
@@ -200,7 +200,7 @@ public RestStatus addEntity(String index, String entityId, JsonNode inputEntity)
             Map<String, Object> inputMap = JSONUtil.convertJsonNodeToMap(inputEntity);
             response = getClient(index).index(new IndexRequest(index, searchType, entityId).source(inputMap), RequestOptions.DEFAULT);
         } catch (IOException e) {
-            logger.error("Exception in adding record to ElasticSearch", e);
+            logger.error("Exception in adding record to ElasticSearch: {}", e.getMessage());
```
Comment (srprasanna) [inline java/elastic-search/src/main/java/dev/sunbirdrc/elastic/ElasticServiceImpl.java:203]: Do this for all errors please.

File: java/elastic-search/src/main/java/dev/sunbirdrc/elastic/ElasticServiceImpl.java [hunk 6]
```diff
@@ -264,7 +264,7 @@ public RestStatus deleteEntity(String index, String osid) {
             readMap.put(Constants.STATUS_KEYWORD, Constants.STATUS_INACTIVE);
             response = getClient(indexL).update(new UpdateRequest(indexL, searchType, osid).doc(readMap), RequestOptions.DEFAULT);
         } catch (NullPointerException | IOException e) {
-            logger.error("exception in deleteEntity {}", e);
+            logger.error("exception in deleteEntity {}", e.getMessage());
```
Comment (srprasanna) [inline java/elastic-search/src/main/java/dev/sunbirdrc/elastic/ElasticServiceImpl.java:267]: either throw the exception or log the stack trace please. Just printing the message would not help anyone.

File: java/middleware-commons/src/test/java/dev/sunbirdrc/registry/middleware/service/ConditionResolverServiceTest.java [hunk 7]
```diff
@@ -44,7 +47,7 @@ public void shouldReturnTrueForValidExpression() throws IOException {
         try {
             resolve = conditionResolverService.resolve(getStudentJsonNode(), requester, condition, attributes);
         } catch (IOException e) {
-            e.printStackTrace();
+            logger.info("Exception occurred: {}", e.getMessage());
```
Comment (srprasanna) [inline java/middleware-commons/src/test/java/dev/sunbirdrc/registry/middleware/service/ConditionResolverServiceTest.java:50]: logger.error and throw the exception

**Review by srprasanna — COMMENTED** [review]
@holashchand  please address the issues I mentioned. Do this for the entire code.

File: docker-compose.yml (hunk 3 above)
Comment (holashchand) [reply docker-compose.yml:220]: we are using 9092 for internal and for outside it's 9094

File: Makefile [hunk 8]
```diff
@@ -28,12 +28,10 @@ java/registry/target/registry.jar: $(SOURCES)
 	cd java && ./mvnw clean install
 
 test: build
-	@echo "VIEW_DIR=java/apitest/src/test/resources/views" >> .env || echo "no permission to append to file"
-	@echo "SCHEMA_DIR=java/apitest/src/test/resources/schemas" >> .env || echo "no permission to append to file"
 	@docker-compose down
 	@rm -rf db-data* || echo "no permission to delete"
 	# test with distributed definition manager and native search
-	@SEARCH_PROVIDER_NAME=dev.sunbirdrc.registry.service.NativeSearchService RELEASE_VERSION=latest KEYCLOAK_IMPORT_DIR=java/apitest/src/test/resources KEYCLOAK_SECRET=a52c5f4a-89fd-40b9-aea2-3f711f14c889 MANAGER_TYPE=DistributedDefinitionsManager DB_DIR=db-data-1 docker-compose up -d db keycloak registry certificate-signer certificate-api redis
+	@docker-compose --env-file test_environments/test1.env up -d db keycloak registry certificate-signer certificate-api redis
```
Comment (srprasanna) [inline Makefile:34]: change the env file name to something meaningful instead of test1

File: Makefile [hunk 9]
```diff
@@ -42,7 +40,7 @@ test: build
 	@docker-compose down
 	@rm -rf db-data-1 || echo "no permission to delete"
 	# test with kafka(async), events, notifications,
-	@NOTIFICATION_ENABLED=true NOTIFICATION_URL=http://notification-ms:8765/notification-service/v1/notification TRACK_NOTIFICATIONS=true EVENT_ENABLED=true ASYNC_ENABLED=true RELEASE_VERSION=latest KEYCLOAK_IMPORT_DIR=java/apitest/src/test/resources KEYCLOAK_SECRET=a52c5f4a-89fd-40b9-aea2-3f711f14c889 DB_DIR=db-data-2 docker-compose up -d db clickhouse redis es keycloak registry certificate-signer certificate-api kafka zookeeper notification-ms metrics
+	@docker-compose --env-file test_environments/test2.env up -d db clickhouse redis es keycloak registry certificate-signer certificate-api kafka zookeeper notification-ms metrics
```
Comment (srprasanna) [inline Makefile:43]: same as above

File: deps/keycloak-mobile-number-login-spi/src/main/java/in/divoc/api/authenticator/NotifyService.java [hunk 10]
```diff
@@ -44,7 +46,7 @@ public void notify(String to, String otp) throws IOException {
 				System.out.println(response.toString());
 			}
 		} catch (IOException e) {
-			logger.error(e);
+			logger.error("Exception occurred while notifying: {}", ExceptionUtils.getStackTrace(e));
```
Comment (srprasanna) [inline deps/keycloak-mobile-number-login-spi/src/main/java/in/divoc/api/authenticator/NotifyService.java:49]: Please throw the IOException here. The signature has throws anyway!

File: java/middleware-commons/src/test/java/dev/sunbirdrc/registry/middleware/service/ConditionResolverServiceTest.java (hunk 7 above)
Comment (srprasanna) [reply java/middleware-commons/src/test/java/dev/sunbirdrc/registry/middleware/service/ConditionResolverServiceTest.java:50]: Never log as log.info . This is an error. We will need the stack trace.  Use log.error and use exception utils for the message. Since the method signature has throws for IOException throw the exception from catch.

File: services/metrics/swagger_gen/restapi/server.go [hunk 11]
```diff
@@ -273,7 +274,7 @@ func (s *Server) Serve() (err error) {
 
 		if s.TLSCACertificate != "" {
 			// include specified CA certificate
-			caCert, caCertErr := os.ReadFile(string(s.TLSCACertificate))
+			caCert, caCertErr := ioutil.ReadFile(string(s.TLSCACertificate))
```
Comment (srprasanna) [inline services/metrics/swagger_gen/restapi/server.go:277]: Why did you change this?

File: services/public-key-service/main.js [hunk 12]
```diff
@@ -37,7 +37,7 @@ const server = http.createServer(async (req, res) => {
 
         console.timeEnd(req.url)
     } catch (e) {
-        console.error(e)
+        console.error(e?.message)
```
Comment (srprasanna) [inline services/public-key-service/main.js:40]: Here you can log e itself. Dont say e.message

File: services/verification-ui/src/components/VerifyCertificate/index.js [hunk 13]
```diff
@@ -29,7 +29,7 @@ export const VerifyCertificate = () => {
         }
     };
     const handleError = err => {
-        console.error(err)
+        console.error(err?.message)
```
Comment (srprasanna) [inline services/verification-ui/src/components/VerifyCertificate/index.js:32]: same as above. We will need the entire stack trace for debugging.

File: services/digilocker-certificate-api/Dockerfile [hunk 14]
```diff
@@ -1,7 +1,7 @@
 ############################
 # STEP 1 build executable binary
 ############################
-FROM golang:alpine AS builder
+FROM golang:1.20-alpine3.18 AS builder
```
Comment (srprasanna) [inline services/digilocker-certificate-api/Dockerfile:4]: why did you change this?

File: services/certificate-signer/main.js [hunk 15]
```diff
@@ -37,7 +37,7 @@ const server = http.createServer(async (req, res) => {
         }
         console.timeEnd(req.url)
     } catch (e) {
-        console.error(e)
+        console.error(e?.message)
```
Comment (srprasanna) [inline services/certificate-signer/main.js:40]: print the stack trace.

File: services/certificate-api/src/routes/certificate_controller.js [hunk 16]
```diff
@@ -299,9 +299,9 @@ async function createPDF(certificate) {
         // close the browser
         await page.close()
         return pdfBuffer
-    } catch (e) {
+    } catch (err) {
         console.log("Failed while creating pdf")
-        console.log(e)
+        console.log(err?.message)
```
Comment (srprasanna) [inline services/certificate-api/src/routes/certificate_controller.js:304]: need stacktrace here.

File: services/certificate-api/Dockerfile [hunk 17]
```diff
@@ -1,9 +1,5 @@
 FROM node:lts-alpine
-RUN apk update && apk add --no-cache nmap && \
-    echo @edge http://nl.alpinelinux.org/alpine/edge/community >> /etc/apk/repositories && \
-    echo @edge http://nl.alpinelinux.org/alpine/edge/main >> /etc/apk/repositories && \
-    apk update && \
-    apk add --no-cache \
+RUN apk add \
```
Comment (srprasanna) [inline services/certificate-api/Dockerfile:2]: WHy did you change this?

File: scripts/index.js [hunk 18]
```diff
@@ -237,7 +237,7 @@ var execute_tasks = function (tasks, fileName, cb) {
             console.log("Executed tasks")
             cb(null)
         } else {
-            console.error(err)
+            console.error(err?.message)
```
Comment (srprasanna) [inline scripts/index.js:240]: Need the full stack trace when there is an error

File: java/view-templates/src/test/java/dev/sunbirdrc/views/ViewTransformerTest.java [hunk 19]
```diff
@@ -76,10 +76,7 @@ private static String readFileContent(String fileName) {
         } catch (FileNotFoundException e1) {
             e1.printStackTrace();
 
-        } catch (IOException e) {
-            e.printStackTrace();
-
-        }
+        } catch (IOException e) {}
```
Comment (srprasanna) [inline java/view-templates/src/test/java/dev/sunbirdrc/views/ViewTransformerTest.java:79]: log stack trace to the logger

File: services/digilocker-certificate-api/Dockerfile (hunk 14 above)
Comment (holashchand) [reply services/digilocker-certificate-api/Dockerfile:4]: build was failing due to latest version on golang:alpine

File: services/certificate-api/Dockerfile (hunk 17 above)
Comment (holashchand) [reply services/certificate-api/Dockerfile:2]: it was causing build failure on runner

File: java/registry/src/main/java/dev/sunbirdrc/registry/app/AppStartupRunner.java [hunk 20]
```diff
@@ -22,6 +22,5 @@ public void run(ApplicationArguments args) throws Exception {
     	logger.info("On Boot starts loading: parent vertex and shard records");
     	entityParenter.ensureKnownParenters();
     	entityParenter.loadDefinitionIndex();
-    	entityParenter.ensureIndexExists();
```
Comment (srprasanna) [inline java/registry/src/main/java/dev/sunbirdrc/registry/app/AppStartupRunner.java:25]: Dont remove this line :)

**Review by srprasanna — APPROVED** [review]

File: services/certificate-api/Dockerfile (hunk 17 above)
Comment (srprasanna) [reply services/certificate-api/Dockerfile:2]: Will there be any issues because you have removed this.

Comment (holashchand) [reply services/certificate-api/Dockerfile:2]: there shouldn't as we have e2e tests which are working fine
